### PR TITLE
Adding docs for upcoming inlineMessage component

### PR DESCRIPTION
**Pull request by lukasoppermann**
Waiting for https://github.com/primer/react/pull/4443 to merge & publish

### CODE REVIEW

File: content/components/inline-message.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    alt="Unavailable InlineMessage"
+    src="https://github.com/primer/design/assets/813754/d0fb73bd-d060-444e-b8c5-0aaff2a54cdc"
+  />
+  <Box as="p" mt="0">
+   <h4>Unavailable</h4>
+    Use unavailable InlineMessages to inform users that a specific part of the UI could not be loaded. For example when not being able to retrieve data in a table.
+  </Box>
+</Box>
+
+### Known accessibility issues (GitHub staff only)
```
Comment (mperrotti) [inline content/components/inline-message.mdx:106]:
```suggestion
## Accessibility

### Known accessibility issues (GitHub staff only)
```

File: content/components/inline-message.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+title: Inline Message
+description: InlineMessage is used to inform the user about the result of an action within the content.
```
Comment (mperrotti) [inline content/components/inline-message.mdx:3]: We use sentence-case for component names every where else.

For example:

- `"InlineMessage"` => `"inline message"`
- `"NavList"` => `"nav list"`

---

I personally prefer PascalCase, but I think it's better to be consistent.

File: content/components/inline-message.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="Critical InlineMessage"
+    src="https://github.com/primer/design/assets/813754/c4190734-b9d2-4978-b14e-6ff0e3716efc"
+  />
+  <Box as="p" mt="0">
+   <h4>Critical</h4>
+    Use critical InlineMessages for errors for example during form validation or when an action failed.
```
Comment (mperrotti) [inline content/components/inline-message.mdx:48]:
```suggestion
    Use the `critical` variant for errors. For example, during form validation or when an action failed.
```

File: content/components/inline-message.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="Critical InlineMessage"
```
Comment (mperrotti) [inline content/components/inline-message.mdx:43]:
```suggestion
    role="presentation"
```

I don't think any of the variant images need alt text.

File: content/components/inline-message.mdx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="Warning InlineMessage"
+    src="https://github.com/primer/design/assets/813754/7bc1679e-812c-41b6-953b-4ef2ffcd9257"
+  />
+  <Box as="p" mt="0">
+   <h4>Warning</h4>
+    Use warning InlineMessages to inform about potential issues or impactful changes. Form example: important settings, possible connectivity concerns.
```
Comment (mperrotti) [inline content/components/inline-message.mdx:66]:
```suggestion
    Use the `warning` variant to inform about potential issues or risky changes. For example, important settings and possible connectivity concerns.
```

File: content/components/inline-message.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="Success InlineMessage"
+    src="https://github.com/primer/design/assets/813754/d3b919f2-4564-43e1-ad31-498d35a09aac"
+  />
+  <Box as="p" mt="0">
+   <h4>Success</h4>
+   Use success InlineMessages to indicate task completion or a successful action. For examples to confirm settings have been saved.
```
Comment (mperrotti) [inline content/components/inline-message.mdx:84]:
```suggestion
   Use the `success` variant to indicate task completion or a successfully completed action. For example, to confirm settings have been saved.
```

File: content/components/inline-message.mdx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="Unavailable InlineMessage"
+    src="https://github.com/primer/design/assets/813754/d0fb73bd-d060-444e-b8c5-0aaff2a54cdc"
+  />
+  <Box as="p" mt="0">
+   <h4>Unavailable</h4>
+    Use unavailable InlineMessages to inform users that a specific part of the UI could not be loaded. For example when not being able to retrieve data in a table.
```
Comment (mperrotti) [inline content/components/inline-message.mdx:102]:
```suggestion
    Use the `unavailable` variant to inform users that a specific part of the UI could not be loaded. For example, when not being able to retrieve data in a table.
```

File: content/components/inline-message.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+## Anatomy
+<img
+  width="960"
+  alt="Anatomy of an InlineMessage, including the icon and the message"
+  src="https://github.com/primer/primitives/assets/813754/d854b989-ff3b-426b-9eec-0355145e534a"
+/>
+
+## Options
+
+### Variants
```
Comment (mperrotti) [inline content/components/inline-message.mdx:32]: You can still write in Markdown if you add returns before and after JSX tags, and don't indent the children of those tags. This gives you doctocat component styles for free, and it wraps headings with anchor links.

For example:

```mdx
<Box
  mb={3}
  display="flex"
  alignItems="flex-start"
  flexDirection={['column', 'column', 'column', 'column', 'row']}
  sx={{gap: 4}}
>

<img
  width="456"
  role="presentation"
  src="https://github.com/primer/design/assets/813754/c4190734-b9d2-4978-b14e-6ff0e3716efc"
/>

<div>

#### Critical

Use the `critical` variant for errors. For example, during form validation or when an action failed.

</div>

</Box>
```

File: src/@primer/gatsby-theme-doctocat/nav.yml [hunk 9]
```diff
@@ -263,6 +263,8 @@
       url: /components/image
     - title: Image crop
       url: /components/image-crop
+    - title: Inline Message
```
Comment (mperrotti) [inline src/@primer/gatsby-theme-doctocat/nav.yml:266]:
```suggestion
    - title: Inline message
```

File: content/components/inline-message.mdx [hunk 10]
```diff
@@ -0,0 +1,108 @@
+---
+title: Inline Message
```
Comment (mperrotti) [inline content/components/inline-message.mdx:2]:
```suggestion
title: Inline message
```

File: content/components/inline-message.mdx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+## Usage
+
+InlineMessages inform users about the result of an action they took. In contrast to a [Banner](./banner) an InlineMessage can be very close the where the action has taken place.
+For example below an input field, next to a button or within a table.
+
+## Anatomy
+<img
+  width="960"
+  alt="Anatomy of an InlineMessage, including the icon and the message"
```
Comment (mperrotti) [inline content/components/inline-message.mdx:26]:
```suggestion
  alt="A labeled diagram of the inline message component. The labeled parts are an icon and a message."
```

**Review by mperrotti — COMMENTED** [review]
These are looking good!

My only feedback is about phrasing and formatting.

File: content/components/inline-message.mdx (hunk 11 above)
Comment (mperrotti) [reply content/components/inline-message.mdx:26]: Alternatively, you could make this `role="presentation"` if you add text that explains each labelled part of the anatomy diagram. See the [tree view](https://primer.style/components/tree-view) for an example.

File: content/components/inline-message.mdx (hunk 2 above)
Comment (lukasoppermann) [inline content/components/inline-message.mdx:3]: /
```suggestion
description: inline message is used to inform the user about the result of an action within the content.
```

**Review by joshblack — COMMENTED** [review]
Just wanted to say looks great! 🔥 

One question I had was for message contents, should these be text-only or is it okay for them to have interactive content?

Also wanted to ask if it is intended for the inline messages to be live region announcements?

**Comment by langermank** [comment]
@lukasoppermann can you add the size option? We have small and medium (default) 

- default 14px, outline icon
- small 12px, filled icon

image

Then I think this is good to go once Josh merges the React PR 😄

**Pull request merged**